draft-barnes-acme.md

```diff
@@ -1291,110 +1291,97 @@ The server provides the following fields as part of the challenge:
+  "accountKey": { "kty": "RSA", ... }
+}
+~~~~~~~~~~
+
+This JWS is NOT REQUIRED to have a "nonce" header parameter (as with the JWS objects that carry ACME request objects).  This allows proof-of-possession response objects to be computed off-line.  For example, as part of a domain transfer, the new domain owner might require the old domain owner to sign a proof-of-possession validation object, so that the new domain owner can present that in an ACME transaction later.
+
+The validation JWS MUST contain a "jwk" header parameter indicating the public key under which the server should verify the JWS.
 
 The client's response includes the server-provided nonce, together with a signature over that nonce by one of the private keys requested by the server.
 
 type (required, string):
 : The string "proofOfPossession"
 
-nonce (required, string):
-: The server nonce that the server previously associated with this challenge
-
-signature (required, object):
-: The ACME signature computed over the signature-input using the server-specified algorithm
-
+authorization (required, JWS):
+: The validation JWS
 
 ~~~~~~~~~~
 
 {
   "type": "proofOfPossession",
-  "nonce": "eET5udtV7aoX8Xl8gYiZIA",
-  "signature": {
-    "alg": "RS256",
-    "nonce": "eET5udtV7aoX8Xl8gYiZIA",
-    "sig": "KxITJ0rNlfDMAtfDr8eAw...fSSoehDFNZKQKzTZPtQ",
-    "jwk": {
-      "kty": "RSA",
-      "e": "AQAB",
-      "n": "KxITJ0rNlfDMAtfDr8eAw...fSSoehDFNZKQKzTZPtQ"
-    }
+  "authorization": {
+    "header": {
+      "alg": "RS256",
+      "jwk": {
+        "kty": "RSA",
+        "e": "AQAB",
+        "n": "AMswMT...3aVtjE"
+      }
+    },
+    "payload": "SfiR1...gSAl7A",
+    "signature": "XcQLfL...cW5beg"
   }
 }
 
 ~~~~~~~~~~
 
-Note that just as in the authorizationRequest message, there are two nonces here, once provided by the client (inside the signature object) and one provided by the server in its challenge (outside the signature object).  The signature covers the concatenation of these two nonces (as specified in the signature-input above).
+To validate a proof-of-possession challenge, the server performs the following steps:
 
-If the server is able to validate the signature and confirm that the jwk and alg objects are unchanged from the original challenge, the server can conclude that the client is in control of the private key that corresponds to the specified public key.  The server can use this evidence in support of its authorization and certificate issuance policies.
+1. Verify that the public key in the "jwk" header of the "authorization" JWS is one of the keys listed in the challenge's "hints" section
+2. Verify the "authorization" JWS using the key indicated in its "jwk" header
+3. Decode the payload of the JWS as UTF-8 encoded JSON
+4. Verify that there are exactly three fields in the decoded object, and that:
+  * The "type" field is set to "proofOfPossession"
+  * The "identifier" field contains the identifier for which authorization is being validated
```

Comment:
All of the signed objects in the challenges have a "type" field indicating the challenge type, and the other ACME signed messages include a "resource" field.  So it should not be ambiguous.